# HG changeset patch
# User Mark Lodato <lodatom@gmail.com>
# Date 1254699547 14400
freeze: Makefile: flags from distutils.sysconfig

Instead of hard-coding the compiler flags and libraries, extract them
from distutils.sysconfig.

diff --git a/Demos/freeze/Makefile b/Demos/freeze/Makefile
index 9f484e4..2b6dd35 100644
--- a/Demos/freeze/Makefile
+++ b/Demos/freeze/Makefile
@@ -1,12 +1,16 @@
 CC = gcc
 CYTHON = ../../bin/cython
 CYTHON_FREEZE = ../../bin/cython_freeze
+PYTHON = python
+
+PY_LDFLAGS = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_var as g; print " ".join([g("LINKFORSHARED"), "-L"+g("LIBPL")])')
+PY_CPPFLAGS = $(shell $(PYTHON) -c 'from distutils.sysconfig import *; print "-I"+get_python_inc()')
+PY_LDLIBS = $(shell $(PYTHON) -c 'from distutils.sysconfig import get_config_var as g; print " ".join(["-lpython"+g("VERSION"), g("SYSLIBS"), g("LIBS"), g("LOCALMODLIBS")])')
 
 CFLAGS = -fPIC -g -O2 -Wall -Wextra
-CPPFLAGS = -I /usr/include/python2.6
-LDFLAGS = -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions
-LDLIBS = /usr/lib/python2.6/config/libpython2.6.a \
-	-lm -ldl -pthread -lutil -lz
+CPPFLAGS = $(PY_CPPFLAGS)
+LDFLAGS = $(PY_LDFLAGS)
+LDLIBS = $(PY_LDLIBS)
 
 
 # Name of executable
-- 
1.6.5.rc3

